### PR TITLE
Fix mismatch between hook events and github event types

### PR DESCRIPTION
**Pull request by Zettat123**
Some workflow trigger events can have multiple activity types, such as `issues` and `pull_request`, and user can specify which types can trigger the workflow. See GitHub documentation: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

Now some hook events cannot match the workflow trigger events correctly because we don't check the activity types. For example, `pull_request_label` is an individual hook event. But there isn't a `pull_request_label` workflow trigger event, we can only use `pull_request` event's `label` activity type. If we don't check the activity types, the workflows without the `label` activity type may be triggered by the `pull_request_label` event by mistake. We need to improve the match logic. 

- [x] [`issues` ](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues)
- [x] [`issue_comment`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issue_comment)
- [x] [`pull_request`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)
- [x] [`pull_request_review`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review) 
- [x] [`pull_request_review_comment`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_review_comment)
- [x] [`release`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release)
- [x] [`registry_package`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#registry_package)

### CODE REVIEW

File: modules/actions/workflows.go [hunk 1]
```diff
@@ -375,3 +408,144 @@ func matchIssueCommentEvent(commit *git.Commit, issueCommentPayload *api.IssueCo
...
+	matchTimes := 0
+	// all acts conditions should be satisfied
+	for cond, vals := range evt.Acts() {
+		switch cond {
+		case "types":
+			action := payload.Action
+			switch action {
+			case api.HookPackageCreated:
+				action = "published"
+				// TODO: support the `updated` activity type
```
Comment (lunny) [inline modules/actions/workflows.go]: deleted activity should be also mentioned in this TODO.

Comment (Zettat123) [reply modules/actions/workflows.go]: I listed the actions with the same name as activity type, the actions need to be converted and the unsupported activity types in the code comments

**Review by lunny — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Comment by yardenshoham** [comment]
Needs manual backport

**Comment by Zettat123** [comment]
Backport lint failed because a function called by unit tests does not exist in v1.19. I add a backport #24091 manually.

**Pull request merged**